src/wallet/wallet.h

```diff
@@ -741,6 +745,106 @@ class CAccountingEntry
     std::vector<char> _ssExtra;
 };
 
+enum NoteType {
+    sapling,
+    sprout
+};
+
+class NotesHelperObject
+{
+public:
+
+    NotesHelperObject(uint256 hash_, uint256 address_, uint64_t timestamp_, NoteType type_,
+        boost::optional<JSOutPoint> jsop_, boost::optional<SaplingOutPoint> op_) :
+    hash(hash_), address(address_), timestamp(timestamp_), type(type_), jsop(jsop_), op(op_)
+    {}
+
+    uint256 hash;
+    uint256 address;
```

Comment:
I'm confused as to how an address can be represented as a `uint256`. What kind of address?

Reply:
This is actually payment address hash but the names are confusing at least so i changed them. `hash` is now `txid` and `address` is `paymentaddress_hash`. Thanks.

https://github.com/zcash/zcash/pull/4399/commits/08e5f2a5f6b1ef6df2438d8723c98397b9ae0c4f